File: app/src/main/java/com/magdenbt/collectionsbenchmark/ui/viewmodelflow/StatViewModel.java
package com.magdenbt.collectionsbenchmark.ui.viewmodelflow;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.magdenbt.collectionsbenchmark.CollectionsType;
import com.magdenbt.collectionsbenchmark.modelflow.StatModel;
import com.magdenbt.collectionsbenchmark.modelflow.StatRepository;

import java.util.List;
import java.util.Objects;

public class StatViewModel extends AndroidViewModel {

    private List<LiveData<StatModel>> statModelsLD;
    private final CollectionsType collectionsType;
    private final StatRepository statRepository;

    public StatViewModel(@NonNull Application application, StatRepository statRepository, CollectionsType collectionsType) {
        super(application);
        this.collectionsType = collectionsType;
        this.statRepository = statRepository;
    }

    public List<LiveData<StatModel>> getStatModelsLD() {
        if (statModelsLD == null) statModelsLD = statRepository.getModels(collectionsType);
        return statModelsLD;
    }

    public void startBenchmark(int sizeCollection, int amountElements) {
        for (LiveData<StatModel> mutableLiveDatum : statModelsLD) {
            try {
                Objects.requireNonNull(mutableLiveDatum.getValue()).startBenchmark(sizeCollection, amountElements);
            } catch (NullPointerException e) {
                Log.e("RVStatViewModel", mutableLiveDatum.toString() + " has NULL value");
            }
        }
    }
}

File: app/src/main/java/com/magdenbt/collectionsbenchmark/ui/viewmodelflow/StatViewModel.kt
package com.magdenbt.collectionsbenchmark.ui.viewmodelflow

import android.content.Context
import androidx.lifecycle.ViewModel
import com.magdenbt.collectionsbenchmark.modelflow.CollectionsType
import com.magdenbt.collectionsbenchmark.modelflow.StatModel
import com.magdenbt.collectionsbenchmark.modelflow.StatRepository

class StatViewModel(
    val context: Context,
) : ViewModel() {

    val statRepository: StatRepository = StatRepository(context)

    val statModelsLD: Map<CollectionsType, List<StatModel>> =
        mutableMapOf<CollectionsType, List<StatModel>>().apply {
            CollectionsType.values().forEach { collectionsType ->
                put(collectionsType, statRepository.getModels(collectionsType))
            }
        }

    fun startBenchmark(collectionsType: CollectionsType, sizeCollection: Int, amountElements: Int) {
        statModelsLD.get(collectionsType)?.iterator()
            ?.forEach { it.startBenchmark(sizeCollection, amountElements) }
    }
}
